crates/subspace-service/src/sync_from_dsn/snap_sync.rs

```diff
@@ -255,28 +335,30 @@ where
         });
     }
 
-    debug!(
-        blocks_count = %blocks.len(),
-        "Queuing importing remaining blocks from last segment"
-    );
-
-    for (_block_number, block_bytes) in blocks {
-        let signed_block = decode_block::<Block>(&block_bytes)
-            .map_err(|error| format!("Failed to decode archived block: {error}"))?;
-        let (header, extrinsics) = signed_block.block.deconstruct();
+    if import_blocks_from_downloaded_segment {
```

Comment:
What is the use case of not doing this though?

Reply:
Can we name it something like `import_a_single_block` or similar? Right now ti indicates we do not import blocks, but we do, we just import one instead of all.